### PR TITLE
Add JS API

### PR DESCRIPTION
**Pull request by Blasz**
This PR splits bolt-check into a CLI and js API so that we can build on top of bolt-check in our own scripts. As part of this I split check and fix into two separate functions to make things more composable, not sure if this was overkill or not.

I also added some integration tests to verify the CLI was printing the correct things.

### CODE REVIEW

File: packages/bolt-check/src/index.ts [hunk 1]
```diff
@@ -1,38 +1,2 @@
...
-    }
-  }
-);
-
-(async () => {
-  // @ts-ignore
-  boltCheck(flags);
-})();
+export { check, fix } from "./api";
+export { default as cli } from "./cli";
```
Comment (Noviny) [inline packages/bolt-check/src/index.ts]: I'm not sure how I feel about this atm - I feel like we might do better having two entrypoints for the two usages, and have the bin reference the cli entrypoint, and api reference this entrypoint.

That said, I can't find the multiple entrypoint docs for preconstruct atm, so this task I am fine if you don't do, leave for later.

File: packages/bolt-check/test.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+yarn-workspace-base-pkg-a relies on get-workspaces at ^0.2.1, but your project relies on  get-workspaces at ^0.2.0.
+`);
+    }
+  });
+
+  it.skip("should print multiple errors for multiple internal mismatches", async () => {
+    throw new Error("write this test");
+  });
+  it.skip("should print multiple errors for multiple external mismatches", async () => {
+    throw new Error("write this test");
```
Comment (Noviny) [inline packages/bolt-check/test.ts]: I'm assuming this is my doing?

File: packages/bolt-check/test.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      __dirname,
+      "yarn-workspace-mismatched-external"
+    );
+    try {
+      await spawn(`${__dirname}/bin.js`, [`--cwd=${cwd}`], {
+        capture: ["stdout", "stderr"]
+      });
+      expect("code should throw").toBe(true);
+    } catch (e) {
+      expect(stripAnsi(e.stderr)).toBe(`there are errors in your config!
```
Comment (Noviny) [inline packages/bolt-check/test.ts]: Thing that would be good here - a test for message when no errors.

**Comment by mitchellhamilton** [comment]
Here's a diff to make the cli an entrypoint:

```diff
diff --git a/packages/bolt-check/bin.js b/packages/bolt-check/bin.js
index af819e1..4a5f635 100755
--- a/packages/bolt-check/bin.js
+++ b/packages/bolt-check/bin.js
@@ -1,6 +1,4 @@
 #!/usr/bin/env node
 "use strict";
 
-const cli = require("./").cli;
-
-cli();
+require("./cli");
diff --git a/packages/bolt-check/package.json b/packages/bolt-check/package.json
index a6d1091..d0d3bee 100644
--- a/packages/bolt-check/package.json
+++ b/packages/bolt-check/package.json
@@ -12,7 +12,8 @@
   },
   "files": [
     "dist",
-    "bin.js"
+    "bin.js",
+    "cli"
   ],
   "dependencies": {
     "@babel/runtime": "^7.4.4",
@@ -28,5 +29,11 @@
     "@types/child-process-promise": "^2.2.1",
     "child-process-promise": "^2.2.1",
     "jest-fixtures": "^0.5.0"
+  },
+  "preconstruct": {
+    "entrypoints": [
+      ".",
+      "cli"
+    ]
   }
 }
diff --git a/packages/bolt-check/src/cli/index.ts b/packages/bolt-check/src/cli/index.ts
index 6cccecd..0f8e5bb 100644
--- a/packages/bolt-check/src/cli/index.ts
+++ b/packages/bolt-check/src/cli/index.ts
@@ -56,8 +56,7 @@ let printErrors = (errors: ErrorObj[]) => {
   });
 };
 
-export default () => {
-  const { flags } = meow(
+const { flags } = meow(
   `
     Usage
       $ bolt-check
@@ -86,9 +85,9 @@ export default () => {
       }
     }
   }
-  );
+);
 
-  (async () => {
+(async () => {
   const config = { cwd: flags.cwd };
   // @ts-ignore
   const errors = await check(config);
@@ -108,5 +107,4 @@ export default () => {
   } else {
     console.log("Looks like your dependencies are fine");
   }
-  })();
-};
+})();
diff --git a/packages/bolt-check/src/index.ts b/packages/bolt-check/src/index.ts
index 56eb3cf..4782867 100644
--- a/packages/bolt-check/src/index.ts
+++ b/packages/bolt-check/src/index.ts
@@ -1,2 +1 @@
 export { check, fix } from "./api";
-export { default as cli } from "./cli";
```

**Pull request merged**